### ISSUE TITLE
[Support]: My recordings are being saved but WHERE!??

### ISSUE THREAD

**Issue by FinnakaLite** [blakeblackshear/frigate#8592]
### Describe the problem you are having

So, I have become absolutly dumbfounded over this, I have 24/7 recording setup, and it works all fine and well, but the files don't exist? Ive checked disk I/O usage when streaming recordings and there's no usage? It is like its pulling recordings out of thin air. Below is a link to a video which explains it well, its difficult to explain so I have it laid out in the video.

[https://streamable.com/f2ksdt](https://streamable.com/f2ksdt)

### Version

0.12.1-367D724

### Frigate config file

```yaml
mqtt:
  enabled: False 
go2rtc:
  streams:
    carport: # <- for RTSP streams
      - [redacted]
      - "ffmpeg:rtsp_cam#audio=opus" # <- copy of the stream which transcodes audio>
    frontdoor: # <- for RTSP streams
      -  [redacted]
      - "ffmpeg:rtsp_cam#audio=opus" # <- copy of the stream which transcodes audio>
#    rtsp_cam2: # <- for RTSP streams
#      - rtsp://zephyr.rtsp.stream/movie?streamKey=fbfbf0cd61c3021847e7ece3d292772f >
#      - "ffmpeg:rtsp_cam#audio=opus" # <- copy of the stream which transcodes audio>

cameras:
  carport:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/carport
          roles:
            - record
            - motion
  frontdoor:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/frontdoor
          roles:
            - record
            - motion
  #bunny:
  #  ffmpeg:
  #    inputs:
  #      - path: rtsp://zephyr.rtsp.stream/movie?streamKey=fbfbf0cd61c3021847e7ece3d2>
  #        roles:
  #          - record

    detect:
      width: 1280
      height: 720

record:
  enabled: True
  retain:
    days: 7
    mode: all
  events:
    retain:
      default: 14
      mode: motion
```

### Relevant log output

```shell
Only issues found in the log are a lot of ffmpeg crashes:

2023-11-12 21:57:10.579454304  [2023-11-12 21:57:10] frigate.video                  ERROR   : frontdoor: Unable to read frames from ffmpeg process.
2023-11-12 21:57:10.579509707  [2023-11-12 21:57:10] frigate.video                  ERROR   : carport: Unable to read frames from ffmpeg process.
2023-11-12 21:57:10.579552668  [2023-11-12 21:57:10] frigate.video                  ERROR   : frontdoor: ffmpeg process is not running. exiting capture thread...
2023-11-12 21:57:10.579590168  [2023-11-12 21:57:10] frigate.video                  ERROR   : carport: ffmpeg process is not running. exiting capture thread...
2023-11-12 21:57:15.456798931  [2023-11-12 21:57:15] watchdog.carport               ERROR   : Ffmpeg process crashed unexpectedly for carport.
2023-11-12 21:57:15.456866528  [2023-11-12 21:57:15] watchdog.frontdoor             ERROR   : Ffmpeg process crashed unexpectedly for frontdoor.
2023-11-12 21:57:15.456902856  [2023-11-12 21:57:15] watchdog.carport               ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-11-12 21:57:15.462414398  [2023-11-12 21:57:15] watchdog.frontdoor             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
```

### FFprobe output from your camera

```shell
Unable to provide
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

TP-Link Tapo C310

### Any other information that may be helpful

The disk it is trying to write to is a RAID 0 array. It is not the boot drive.

**Comment by NickM-27** [comment]
It is wherever you mapped to /media/frigate in the docker config. Also worth mentioning that `motion` is not a valid role

**Comment by FinnakaLite** [comment]
> It is wherever you mapped to /media/frigate in the docker config. Also worth mentioning that `motion` is not a valid role

@NickM-27 The thing is you see it simply ISNT? I don’t know why or how, if you look in the video, about half way through, it’s linked but isn’t there. 

Also I know motion isn’t valid, I was messing around with it a while back and just forgot to remove it.

**Comment by NickM-27** [comment]
ah for some reason the video stopped playing halfway through, I had to manually skip ahead this time. Anyway, it is possible this docker stack has not been deployed yet and the files are being recorded to /var in the docker folder. 

I would also highly suggest leaving the tmpfs enabled for /tmp/cache

**Comment by Sch0field9** [comment]
seems like you are trying ls in ~/disk/r1/... instead of /disk/r1...
your directory might look like this 
- docker:/disk/r1/frigate-media/actual-recordings.mp4
- docker:/root/disk/r1/frigate-media/(empty) 
  (also shown as ~/disk/r1/...)

did you try:
```shell
cd /.
``` 
?

and looked if there is your "disk/r1" directory?

**Comment by FinnakaLite** [comment]
> seems like you are trying ls in ~/disk/r1/... instead of /disk/r1... your directory might look like this
> 
> * docker:/disk/r1/frigate-media/actual-recordings.mp4
> * docker:/root/disk/r1/frigate-media/(empty)
>   (also shown as ~/disk/r1/...)
> 
> did you try:
> 
> ```shell
> cd /.
> ```
> 
> ?
> 
> and looked if there is your "disk/r1" directory?

Nope, im looking in the correct place, double checked the other one aswell and that directory does not exist, its a RAID array mounted at /disk/r1/

**Comment by FinnakaLite** [comment]
> ah for some reason the video stopped playing halfway through, I had to manually skip ahead this time. Anyway, it is possible this docker stack has not been deployed yet and the files are being recorded to /var in the docker folder.
> 
> I would also highly suggest leaving the tmpfs enabled for /tmp/cache

Should I try letting it run for a few days and see if it sorts itself out?

**Comment by FinnakaLite** [comment]
Update: Ran `find / -type d -name "frontdoor"` on my system, since I know frigate structures recordings under folders named the same as the cameras, and found nothing on my entire system... Ran the same with the other names of my cameras and ran a "find" for "recordings"

**Comment by Trashmee** [comment]
Did you jupm into the container & check folder content there?
Like:
```
docker exec -it frigate bash
ls -lash /media/frigate
df -h /media/frigate
```` 

The later one should give you an indication on the capacity of the disk data is stored to.
8GB of total space (shown in Frigate WebUI -> storage) doesn't seem a lot.

**Comment by FinnakaLite** [comment]
I finally fixed the issue after over two months of trying. Here's what I've learned: If you want to have your recording files or any files anywhere but your home folder, DO NOT USE SNAP DOCKER. 

When using snap docker instead of apt docker, containers can only access files in your $HOME folder. 

I would recommend that @blakeblackshear or whoever is responsible for documentation add this as a note on the installation page.

To be honest, don't use snap docker; I can't see any reason good why anymore.
